mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -901,23 +946,57 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
         setChatHistoryManager(chatHistoryManager)
     }
 
-    const markAgentForStopping = (): void => {
-        // Signal that the agent should stop after current task
+    const markAgentForStopping = async (): Promise<void> => {
+        // 1. Immediately abort any ongoing requests
+        if (activeRequestControllerRef.current) {
+            activeRequestControllerRef.current.abort();
+            activeRequestControllerRef.current = null;
+        }
+        
+        // 2. Disconnect any active stream handlers
+        if (streamHandlerRef.current) {
+            websocketClient.stream.disconnect(streamHandlerRef.current, null);
+            streamHandlerRef.current = null;
+        }
+        
+        // 3. Clear streaming content
+        streamingContentRef.current = '';
+        
+        // 4. Signal that the agent should stop immediately
         shouldContinueAgentExecution.current = false;
-        // Update UI to show stopping state
-        setAgentExecutionStatus('stopping');
-    }
+        
+        // 5. Update UI to show immediate stop (not "stopping")
+        setAgentExecutionStatus('idle');
+        setLoadingAIResponse(false);
 
-    const finalizeAgentStop = (): void => {
-        // Notify user that agent has been stopped
-        shouldContinueAgentExecution.current = false;
+        // 6. Add immediate feedback message
         const newChatHistoryManager = getDuplicateChatHistoryManager();
         addAIMessageFromResponseAndUpdateState(
-            "Agent execution stopped. You can continue the conversation or start a new one.",
+            "Agent stopped by user.",
             'chat',
             newChatHistoryManager
         );
-        // Reset agent to idle state
+
+        // 7. Send stop message to backend
+        try {
+            await websocketClient.sendMessage({
+                type: "stop_agent",
+                message_id: UUID.uuid4(),
+                metadata: {
+                    promptType: "stop_agent",
+                    threadId: activeThreadIdRef.current
+                },
+                stream: false
+            });
+        } catch (error) {
+            console.error('Failed to send stop agent message to backend:', error);
+        }
+    }
+
+    const finalizeAgentStop = (): void => {
+        // This function is now handled by markAgentForStopping for immediate feedback
+        // Keeping it for backward compatibility with existing code that calls it
+        shouldContinueAgentExecution.current = false;
```

Comment:
If we're not using this function anymore, then we can remove it. There's no backwards compatibility issues that we need to be worried about here.

Reply:
But it does look like we are using it throughout the codebase still. What does this comment mean?